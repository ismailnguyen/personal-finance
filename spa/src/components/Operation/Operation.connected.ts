import { OperationComponent, OwnProps, StateProps, DispatchProps } from "./Operation";
import { connect } from "react-redux";
import { ExtendedDispatch } from "../../business/definitions";
import { ApplicationState } from "../../business/state";
import { Operation as IOperation } from "../../business/account/operation";
import business from "../../business";

function mapStateToProps(state: ApplicationState, { id }: OwnProps): StateProps {
  const operation: IOperation = business.getOperation(state, id);
  return { operation };
}

function mapDispatchToProps(dispatch: ExtendedDispatch, { id }: OwnProps): DispatchProps {
  const deleteOperation = () => {
    const thunk = business.deleteOperation(id);
    dispatch(thunk);
  };
  return { deleteOperation };
}

export const Operation = connect<StateProps, DispatchProps, OwnProps, ApplicationState>(
  mapStateToProps,
  mapDispatchToProps
)(OperationComponent);
