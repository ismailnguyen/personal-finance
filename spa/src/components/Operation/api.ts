import { _delete } from "../../util/xhr";

export function deleteOperation(id: string): Promise<void> {
  return _delete("/operation/" + id);
}
