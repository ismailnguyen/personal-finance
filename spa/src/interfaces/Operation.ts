import { Omit } from "../util/omit";

export interface Operation {
  id: string;
  date: Date;
  amount: number;
}

export type SerializedOperation = Omit<Operation, "date"> & {
  date: string;
};
