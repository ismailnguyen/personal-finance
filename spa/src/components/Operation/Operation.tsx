import React from "react";
import { Operation as IOperation } from "../../business/operation/model";
import { FinanceDate } from "../FinanceDate";
import { LabeledDebitOrCredit } from "../LabeledDebitOrCredit";
import { OperationActions } from "../OperationActions/OperationActions";
import "./Operation.scss";

export interface OwnProps {
  id: string;
}

export interface StateProps {
  operation: IOperation;
}

export interface DispatchProps {
  requestDeleteOperation(): void;
}

export type Props = OwnProps & StateProps & DispatchProps;

export const OperationComponent: React.StatelessComponent<Props> = ({
  operation: { date, amount },
  requestDeleteOperation
}) => (
  <div className="operation" data-e2e="account-operation">
    <LabeledDebitOrCredit amount={amount} renderLabel={() => <FinanceDate date={date} />} />
    <div className="actions-container">
      <div className="actions-content">
        <OperationActions onDelete={requestDeleteOperation} />
      </div>
    </div>
  </div>
);
