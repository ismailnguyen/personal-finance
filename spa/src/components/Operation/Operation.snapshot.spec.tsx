import React from "react";
import renderer from "react-test-renderer";
import { OperationComponent, Props } from "./Operation";

it("renders correctly", () => {
  const props: Props = {
    id: "0",
    operation: {
      id: "0",
      date: new Date(2019, 4, 24),
      amount: 10
    },
    deleteOperation: jest.fn()
  };
  const tree = renderer.create(<OperationComponent {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
