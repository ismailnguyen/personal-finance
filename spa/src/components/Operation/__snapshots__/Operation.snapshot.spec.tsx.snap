// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="operation"
  data-e2e="account-operation"
>
  <div
    className="labeled-debit-or-credit"
  >
    <div
      className="label-container"
    >
      <span
        className="date"
      >
        2019-5-24
      </span>
    </div>
    <div
      className="amount-container"
    >
      <div
        className="debit-or-credit"
      >
        <span
          className="credit"
        >
          <span
            className="value"
          >
            +10
          </span>
        </span>
        <span
          className="debit"
        />
      </div>
    </div>
  </div>
  <div
    className="actions-container"
  >
    <div
      className="actions-content"
    >
      <div
        className="actions"
      >
        <button
          className="delete mdl-button mdl-js-button mdl-button--icon"
          data-e2e="account-operation-delete"
          onClick={[MockFunction]}
        >
          <i
            className="material-icons"
          >
            delete
          </i>
        </button>
      </div>
    </div>
  </div>
</div>
`;
